controllers/scopeinstance_controller.go

```diff
@@ -104,210 +131,167 @@ func (r *ScopeInstanceReconciler) Reconcile(ctx context.Context, req ctrl.Reques
-					APIGroup: "rbac.authorization.k8s.io",
-				},
-			}
+	u := r.patchConfigForClusterRoleBinding(existingCRB, crb)
 
-			crbList := &rbacv1.ClusterRoleBindingList{}
-			if err := r.Client.List(ctx, crbList, client.MatchingLabels{
-				scopeInstanceUIDKey:           string(in.GetUID()),
-				scopeTemplateUIDKey:           string(st.GetUID()),
-				clusterRoleBindingGenerateKey: cr.GenerateName,
-			}); err != nil {
-				return err
-			}
+	// server-side apply patch
+	if err := r.patchBinding(ctx, u); err != nil {
+		return err
+	}
 
-			if len(crbList.Items) > 1 {
-				return fmt.Errorf("more than one ClusterRoleBinding found for ClusterRole %s", cr.GenerateName)
-			}
+	return nil
+}
 
-			// GenerateName is immutable, so create the object if it has changed
-			if len(crbList.Items) == 0 {
-				if err := r.Client.Create(ctx, crb); err != nil {
-					return err
-				}
-				continue
-			}
+func (r *ScopeInstanceReconciler) patchConfigForClusterRoleBinding(oldCrb *rbacv1.ClusterRoleBinding, crb *rbacv1.ClusterRoleBinding) *unstructured.Unstructured {
+	return &unstructured.Unstructured{
+		Object: map[string]interface{}{
+			"apiVersion": rbacv1.SchemeGroupVersion.String(),
+			"kind":       "ClusterRoleBinding",
+			"metadata": map[string]interface{}{
+				"name":            oldCrb.Name,
+				"ownerReferences": crb.OwnerReferences,
+				"labels":          crb.Labels,
+			},
+			"subjects": crb.Subjects,
+		},
+	}
+}
 
-			existingCRB := &crbList.Items[0]
+func (r *ScopeInstanceReconciler) createOrUpdateRoleBinding(ctx context.Context, cr *operatorsv1.ClusterRoleTemplate, in *operatorsv1.ScopeInstance, st *operatorsv1.ScopeTemplate, namespace string) error {
+	rb := r.getRoleBinding(cr, in, st, namespace)
+	rbList := &rbacv1.RoleBindingList{}
+	if err := r.Client.List(ctx, rbList, &client.ListOptions{
+		Namespace: namespace,
+	}, client.MatchingLabels{
+		scopeInstanceUIDKey:           string(in.GetUID()),
+		clusterRoleBindingGenerateKey: cr.GenerateName,
+	}); err != nil {
+		return err
+	}
 
-			if util.IsOwnedByLabel(existingCRB.DeepCopy(), in) &&
-				reflect.DeepEqual(existingCRB.Subjects, crb.Subjects) &&
-				reflect.DeepEqual(existingCRB.Labels, crb.Labels) {
-				log.Log.Info("existing ClusterRoleBinding does not need to be updated")
-				return nil
-			}
-			existingCRB.Labels = crb.Labels
-			existingCRB.OwnerReferences = crb.OwnerReferences
-			existingCRB.Subjects = crb.Subjects
-
-			// server-side apply patch
-			existingCRB.ManagedFields = nil
-			if err := r.Client.Patch(ctx,
-				existingCRB,
-				client.Apply,
-				client.FieldOwner(siCtrlFieldOwner),
-				client.ForceOwnership); err != nil {
-				return err
-			}
+	if len(rbList.Items) > 1 {
+		return fmt.Errorf("more than one RoleBinding found for ClusterRole %s", cr.GenerateName)
+	}
 
+	// GenerateName is immutable, so create the object if it has changed
+	if len(rbList.Items) == 0 {
+		if err := r.Client.Create(ctx, rb); err != nil {
+			return err
 		}
```

Comment:
So I looked into doing this and I ran into an issue where I couldn't find a generic interface/object for a list with a `.Items` field or `GetItems` function that could be reasonably checked. I even looked into Go generics but couldn't get that to work properly. My attempts at making this a more generic function resulted in more unreadable code to do the same thing. 

I can take another stab at it, as I didn't spend too long trying to get the generic function to work.

Reply:
+1 to what @everettraven said, it's hard to make them generic. I wasn't able to reconcile how to make them generic. I tried to see if it was possible.

Reply:
So there was some more conversation regarding this during scrum and @joelanford mentioned a potential solution for creating a modular solution that would allow for more generalized logic that could apply to all the resources we want to create/update. That being said, we also discussed that it would be best to do this in a separate issue/PR to keep the scope and size of this PR smaller.